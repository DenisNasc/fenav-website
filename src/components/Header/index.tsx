import React, {useState} from 'react';
import {makeStyles, createStyles, Theme} from '@material-ui/core/styles';
import {AppBar, Container, Tabs, Tab, Typography} from '@material-ui/core';

import {ReactComponent as Logo} from '../../assets/icons/FENAV.svg';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    header: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      height: '100%',
    },
    container: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-start',
      width: '400px',
      padding: '0px',
      paddingLeft: '20px',
      margin: '0px',
    },
    title: {
      marginLeft: '60px',
    },
    logo: {
      paddingBottom: '5px',
      paddingLeft: '20px',
      width: '60px',
      height: '80px',
    },
    categoriesContainer: {
      height: '100%',
      display: 'flex',
      paddingRight: '20px',
      alignItems: 'center',
    },
    categorie: {
      padding: '0px',
      height: '80px',
      marginRight: '5px',
    },
  })
);

const Header = () => {
  const a11yProps = (index: any) => {
    return {
      id: `simple-tab-${index}`,
      'aria-controls': `simple-ItemHeader-${index}`,
    };
  };

  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue);
  };

  const [value, setValue] = useState(0);
  const classes = useStyles();

  return (
    <AppBar data-testid="Header" className={classes.header} position="static">
      <Container classes={{root: classes.container}}>
        <Logo className={classes.logo} />
        <Typography classes={{root: classes.title}} data-testid="header-title">
          Faculdade de Engenharia Naval
        </Typography>
      </Container>
      <Tabs
        classes={{root: classes.categoriesContainer}}
        value={value}
        onChange={handleChange}
        aria-label="simple tabs example"
      >
        <Tab
          data-testid="categorie-1"
          classes={{root: classes.categorie}}
          label="A Faculdade"
          {...a11yProps(0)}
        />
        <Tab
          data-testid="categorie-2"
          classes={{root: classes.categorie}}
          label="Como Ingressar"
          {...a11yProps(1)}
        />
        <Tab
          data-testid="categorie-3"
          classes={{root: classes.categorie}}
          label="Notícias"
          {...a11yProps(2)}
        />
        <Tab
          data-testid="categorie-4"
          classes={{root: classes.categorie}}
          label="Publicações"
          {...a11yProps(3)}
        />
        <Tab
          data-testid="categorie-5"
          classes={{root: classes.categorie}}
          label="Pós-Graduação"
          {...a11yProps(4)}
        />
        <Tab
          data-testid="categorie-6"
          classes={{root: classes.categorie}}
          label="Área do Aluno"
          {...a11yProps(5)}
        />
      </Tabs>
    </AppBar>
  );
};

export default Header;
