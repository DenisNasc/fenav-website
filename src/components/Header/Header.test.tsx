import React from 'react';
import ReactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '.';

describe('Header Component', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  test('should match to the faculty name', () => {
    const {getByTestId} = render(<Header />);
    expect(getByTestId('header-title')).toHaveTextContent(/Faculdade de Engenharia Naval/);
  });

  describe('categories buttons', () => {
    test('should match to the categorie 1 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-1')).toHaveTextContent(/A Faculdade/);
    });

    test('should match to the categorie 2 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-2')).toHaveTextContent(/Como Ingressar/);
    });

    test('should match to the categorie 3 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-3')).toHaveTextContent(/Notícias/);
    });

    test('should match to the categorie 4 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-4')).toHaveTextContent(/Publicações/);
    });

    test('should match to the categorie 5 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-5')).toHaveTextContent(/Pós-Graduação/);
    });

    test('should match to the categorie 6 name', () => {
      const {getByTestId} = render(<Header />);
      expect(getByTestId('categorie-6')).toHaveTextContent(/Área do Aluno/);
    });
  });

  test('should match to the snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
