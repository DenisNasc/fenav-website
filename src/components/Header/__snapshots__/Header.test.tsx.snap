// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match to the snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-1 MuiPaper-elevation4"
  data-testid="Header"
>
  <div
    className="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
  >
    <svg
      className="makeStyles-logo-4"
    >
      FENAV.svg
    </svg>
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
      data-testid="header-title"
    >
      Faculdade de Engenharia Naval
    </p>
  </div>
  <div
    className="MuiTabs-root makeStyles-categoriesContainer-5"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="simple tabs example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-controls="simple-ItemHeader-0"
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit Mui-selected"
          data-testid="categorie-1"
          disabled={false}
          id="simple-tab-0"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            A Faculdade
          </span>
          <span
            className="PrivateTabIndicator-root-108 PrivateTabIndicator-colorSecondary-110 MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-controls="simple-ItemHeader-1"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit"
          data-testid="categorie-2"
          disabled={false}
          id="simple-tab-1"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Como Ingressar
          </span>
        </button>
        <button
          aria-controls="simple-ItemHeader-2"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit"
          data-testid="categorie-3"
          disabled={false}
          id="simple-tab-2"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Notícias
          </span>
        </button>
        <button
          aria-controls="simple-ItemHeader-3"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit"
          data-testid="categorie-4"
          disabled={false}
          id="simple-tab-3"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Publicações
          </span>
        </button>
        <button
          aria-controls="simple-ItemHeader-4"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit"
          data-testid="categorie-5"
          disabled={false}
          id="simple-tab-4"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Pós-Graduação
          </span>
        </button>
        <button
          aria-controls="simple-ItemHeader-5"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root makeStyles-categorie-6 MuiTab-textColorInherit"
          data-testid="categorie-6"
          disabled={false}
          id="simple-tab-5"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Área do Aluno
          </span>
        </button>
      </div>
    </div>
  </div>
</header>
`;
